requests/extension-requests/extensionRequests.js

```diff
@@ -0,0 +1,1211 @@
+  clearCheckboxes('status-filter');
+  filterModal.classList.toggle('hidden');
+  changeFilter();
+  updateFilterStates('status', '');
+  updateUrl();
+  await populateExtensionRequests(filterStates);
+});
+
+filterModal.addEventListener('click', (event) => {
+  event.stopPropagation();
+});
+
+window.onclick = function () {
+  filterModal.classList.add('hidden');
+};
+
+searchElement.addEventListener('keypress', async (event) => {
+  if (event.key === 'Enter') {
+    const usernames = event.target.value.trim();
+    let userIdList = [];
+    if (usernames) {
+      const usersList = await processUsernames(usernames);
+      userIdList = usersList.userIdList;
+      assigneeUsernamesList = usersList.validUsernameList;
+      if (userIdList.length === 0) {
+        searchElement.setCustomValidity('No users found!');
+        searchElement.reportValidity();
+        return;
+      }
+    } else {
+      assigneeUsernamesList = [];
+    }
+    const assigneeFilterState = userIdList.length ? userIdList : '';
+    updateFilterStates('assignee', assigneeFilterState);
+    updateUrl();
+    changeFilter();
+    await populateExtensionRequests(filterStates);
+  }
+});
+
+sortButton.addEventListener('click', async (event) => {
+  toggleTooltipText();
+  toggleSortIcon();
+  toggleOrder();
+  changeFilter();
+  await populateExtensionRequests(filterStates);
+});
+
+const toggleTooltipText = () => {
+  const tooltip = sortButton.querySelector('.tooltip');
+  if (tooltip.textContent === 'Oldest first') {
+    tooltip.textContent = 'Newest first';
+  } else {
+    tooltip.textContent = 'Oldest first';
+  }
+};
+
+const toggleOrder = () => {
+  if (filterStates.order === Order.DESCENDING) {
+    updateFilterStates('order', Order.ASCENDING);
+  } else {
+    updateFilterStates('order', Order.DESCENDING);
+  }
+  updateUrl();
+};
+
+const toggleSortIcon = () => {
+  if (ascIcon.style.display === 'none') {
+    descIcon.style.display = 'none';
+    ascIcon.style.display = 'block';
+  } else {
+    descIcon.style.display = 'block';
+    ascIcon.style.display = 'none';
+  }
+};
+
+filterButton.addEventListener('click', (event) => {
+  event.stopPropagation();
+  filterModal.classList.toggle('hidden');
+});
+
+populateStatus();
+
+const handleFormPropagation = async (event) => {
+  event.preventDefault();
+};
+
+async function createExtensionCard(data) {
+  renderLogRecord[data.id] = [];
+  //Create card element
+  const rootElement = createElement({
+    type: 'div',
+    attributes: { class: 'extension-card' },
+  });
+  extensionRequestsContainer.appendChild(rootElement);
+  const removeSpinner = addSpinner(rootElement);
+  rootElement.classList.add('disabled');
+  //Api calls
+  const userDataPromise = getUser(data.assignee);
+  const taskDataPromise = getTaskDetails(data.taskId);
+  const isDeadLineCrossed = Date.now() > secondsToMilliSeconds(data.oldEndsOn);
+  const isNewDeadLineCrossed =
+    Date.now() > secondsToMilliSeconds(data.newEndsOn);
```

Comment:
In the `createExtensionCard` function, there are multiple `document.createElement` calls. This can be optimized by using a document fragment or template literals.